### PR TITLE
unified RBAC: add EM in v1.0

### PR DESCRIPTION
**Pull request by markwahl-msft**
Currently the beta unified RBAC docs mentions directory and entitlement management as providers across all the relevant APIs.  The v1.0 mention entitlement management as an application, but not in the API calls for role definitions and role assignments.  Today those v1.0 only mention the directory provider.  This adds the examples in the API with the existing delegated permission scopes (same as we had in beta), for customers to query, create and delete role assignments for entitlement management in v1.0.

### CODE REVIEW

File: api-reference/v1.0/api/rbacapplication-list-roleassignments.md [hunk 1]
```diff
@@ -33,9 +49,21 @@ GET /roleManagement/directory/roleAssignments?$filter=principalId eq '{principal
...
+
+GET /roleManagement/entitlementManagement/roleAssignments?$filter=roleDefinitionId eq '{roleDefinition id}'
+
+GET /roleManagement/entitlementManagement/roleAssignments?$filter=appScopeId eq '/AccessPackageCatalog/{catalog id}'
+```
+
 ## Query parameters
 
-This operation requires the `$filter` query parameter to query specific instances of role assignments. You can filter on the `roleDefinitionId` or `principalId` properties. The `roleDefinitionId` property can be either a role object ID or a **templateId**. For general information, see [OData query parameters](/graph/query-parameters).
+This operation requires the `$filter` query parameter to query specific instances of role assignments. You can filter on the `roleDefinitionId` or `principalId` properties, or for the entitlement management provider, the `appScopeId` property. For the directory provider, the `roleDefinitionId` property can be either a role object ID or a **templateId**. For general information, see [OData query parameters](/graph/query-parameters).
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/rbacapplication-list-roleassignments.md:66]: Is this accurate?
```suggestion
This operation requires the `$filter` query parameter to query role assignments for the supported RBAC providers.
+ For the directory provider, you must filter on either the **roleDefinitionId** or **principalId** properties. The **roleDefinitionId** property can be either a role object ID or a value for the **templateId** property.
+ For the entitlement management provider, you must filter by the **appScopeId** property. 

For general information, see [OData query parameters](/graph/query-parameters).
```

File: api-reference/v1.0/api/rbacapplication-list-roleassignments.md [hunk 2]
```diff
@@ -257,6 +285,55 @@ Content-type: application/json
 }
 ```
 
+### Example 3: Request using $filter for role assignments on an access package catalog and expand principal
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/rbacapplication-list-roleassignments.md:288]:
```suggestion
### Example 3: Request using $filter for role assignments on an access package catalog and expand the principal object
```

File: api-reference/v1.0/api/rbacapplication-list-roledefinitions.md [hunk 3]
```diff
@@ -13,24 +13,47 @@ Namespace: microsoft.graph
...
-One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+Depending on the RBAC provider and the permission type (delegated or application) that is needed, choose from the following tables the least privileged permission required to call this API. To learn more, including [taking caution](/graph/auth/auth-concepts#best-practices-for-requesting-permissions) before choosing more privileged permissions, see [Permissions](/graph/permissions-reference).
+
+### For the directory (Azure AD) provider
 
 |Permission type      | Permissions (from least to most privileged)              |
 |:--------------------|:---------------------------------------------------------|
 |Delegated (work or school account) | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory, Directory.ReadWrite.All    |
 |Delegated (personal Microsoft account) | Not supported.    |
 |Application | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory, Directory.ReadWrite.All |
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/rbacapplication-list-roledefinitions.md:30]: - I support your goal with the updated intro statement. However, for consistency right now, I suggest we maintain the current boilerplate text. There's an initiative to have a repo-wide update to this text so we can advise customers about best practices like you've intended.
- Same request about Directory.ReadWrite unless there are objections.

```suggestion
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

### For the directory (Azure AD) provider

|Permission type      | Permissions (from least to most privileged)              |
|:--------------------|:---------------------------------------------------------|
|Delegated (work or school account) | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory    |
|Delegated (personal Microsoft account) | Not supported.    |
|Application | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory |
```

File: api-reference/v1.0/api/rbacapplication-post-roleassignments.md [hunk 4]
```diff
@@ -15,22 +15,44 @@ Create a new [unifiedRoleAssignment](../resources/unifiedroleassignment.md) obje
 
 ## Permissions
 
-One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+Depending on the RBAC provider and the permission type (delegated or application) that is needed, choose from the following table the least privileged permission required to call this API. To learn more, including [taking caution](/graph/auth/auth-concepts#best-practices-for-requesting-permissions) before choosing more privileged permissions, search for the following permissions in [Permissions](/graph/permissions-reference).
+
+### For the Directory (Azure AD) provider
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/rbacapplication-post-roleassignments.md:21]: Same comment
```suggestion

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

### For the directory (Azure AD) provider
```

File: api-reference/v1.0/api/rbacapplication-post-roleassignments.md [hunk 5]
```diff
@@ -293,6 +315,53 @@ Content-type: application/json
 }
 ```
 
+### Example 4: Create a role assignment at access package catalog scope
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/rbacapplication-post-roleassignments.md:318]:
```suggestion
### Example 4: Create a role assignment with access package catalog scope
```

File: api-reference/v1.0/api/unifiedroleassignment-delete.md [hunk 6]
```diff
@@ -15,22 +15,43 @@ Delete a [unifiedRoleAssignment](../resources/unifiedRoleAssignment.md) object.
...
+
+### For the Directory (Azure AD) provider
 
 | Permission type                        | Permissions (from least to most privileged) |
 |:---------------------------------------|:--------------------------------------------|
 | Delegated (work or school account)     | RoleManagement.ReadWrite.Directory |
 | Delegated (personal Microsoft account) | Not supported. |
 | Application                            | RoleManagement.ReadWrite.Directory |
 
+### For the Entitlement management provider
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/unifiedroleassignment-delete.md:29]: Same comment
```suggestion
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

### For the directory (Azure AD) provider

| Permission type                        | Permissions (from least to most privileged) |
|:---------------------------------------|:--------------------------------------------|
| Delegated (work or school account)     | RoleManagement.ReadWrite.Directory |
| Delegated (personal Microsoft account) | Not supported. |
| Application                            | RoleManagement.ReadWrite.Directory |

### For the entitlement management provider
```

File: api-reference/v1.0/api/unifiedroleassignment-get.md [hunk 7]
```diff
@@ -15,22 +15,42 @@ Retrieve the properties and relationships of a [unifiedRoleAssignment](../resour
...
-One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+Depending on the RBAC provider and the permission type (delegated or application) that is needed, choose from the following table the least privileged permission required to call this API. To learn more, including [taking caution](/graph/auth/auth-concepts#best-practices-for-requesting-permissions) before choosing more privileged permissions, search for the following permissions in [Permissions](/graph/permissions-reference).
+
+### For the Directory (Azure AD) provider
 
 |Permission type      | Permissions (from least to most privileged)              |
 |:--------------------|:---------------------------------------------------------|
 |Delegated (work or school account) | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory, Directory.ReadWrite.All    |
 |Delegated (personal Microsoft account) | Not supported.    |
 |Application | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory, Directory.ReadWrite.All |
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/unifiedroleassignment-get.md:26]: Same comment, same request
```suggestion
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

### For the directory (Azure AD) provider

|Permission type      | Permissions (from least to most privileged)              |
|:--------------------|:---------------------------------------------------------|
|Delegated (work or school account) | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory   |
|Delegated (personal Microsoft account) | Not supported.    |
|Application | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory |
```

File: api-reference/v1.0/api/unifiedroledefinition-get.md [hunk 8]
```diff
@@ -11,26 +11,51 @@ doc_type: "apiPageType"
...
-One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+Depending on the RBAC provider and the permission type (delegated or application) that is needed, choose from the following tables the least privileged permission required to call this API. To learn more, including [taking caution](/graph/auth/auth-concepts#best-practices-for-requesting-permissions) before choosing more privileged permissions, see [Permissions](/graph/permissions-reference).
+
+### For the directory (Azure AD) provider
 
 |Permission type      | Permissions (from least to most privileged)              |
 |:--------------------|:---------------------------------------------------------|
 |Delegated (work or school account) | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory, Directory.ReadWrite.All    |
 |Delegated (personal Microsoft account) | Not supported.    |
 |Application | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory, Directory.ReadWrite.All |
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/unifiedroledefinition-get.md:31]: Same comment, same suggestion
```suggestion
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

### For the directory (Azure AD) provider

|Permission type      | Permissions (from least to most privileged)              |
|:--------------------|:---------------------------------------------------------|
|Delegated (work or school account) | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory    |
|Delegated (personal Microsoft account) | Not supported.    |
|Application | RoleManagement.Read.Directory, Directory.Read.All, RoleManagement.ReadWrite.Directory |
```

File: api-reference/v1.0/resources/entitlementmanagement-overview.md [hunk 9]
```diff
@@ -26,6 +26,8 @@ The entitlement management resource types include:
 - [entitlementManagementSettings](entitlementmanagementsettings.md): Tenant-wide settings for Azure AD entitlement management.
 - [approval](approval.md): represents the decisions associated with an access package request.
 
+In addition, role assignments for entitlement management-specific roles can be managed through entitlement management [role definitions](unifiedroledefinition.md).
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/entitlementmanagement-overview.md:29]:
```suggestion
In addition, you can manage role assignments for entitlement management-specific roles through entitlement management [role definitions](unifiedroledefinition.md).
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @markwahl-msft for closing this gap in the v1.0 docs. I've left a few suggestions for consideration.

File: api-reference/v1.0/api/rbacapplication-list-roledefinitions.md (hunk 3 above)
Comment (markwahl-msft) [reply api-reference/v1.0/api/rbacapplication-list-roledefinitions.md]: Sure, I am rolling back the intro statement. I would suggest though that we not introduce changes to the directory permissions list in *this* PR, as I was intending this PR to document the EM provider.  I'd prefer that changes to the directory provider behavior be orchestrated by @abhijeetsinha or his colleagues in a distinct PR.

Comment (FaithOmbongi) [reply api-reference/v1.0/api/rbacapplication-list-roledefinitions.md]: gotcha. Not a problem - we'll handle that through a distinct PR.

**Review by FaithOmbongi — APPROVED** [review]
Thank you @markwahl-msft for raising this PR to close this doc gap. Approved.

**Pull request merged**